### PR TITLE
Generators e2e tests

### PR DESCRIPTION
**Pull request by ggawryal**

### CODE REVIEW

File: tests/generators/local.tests [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// INTEGRATION TESTS IGNORE
```
Comment (ignacy123) [inline tests/generators/local.tests:1]: This should be at the top of the file with code, not this one

File: tests/generators/stacked_foreach.bzz [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            jeśli(y > m) przerwij
+            otrzymując z: Liczba od przedziału(m+1) {
+                otrzymując w: Liczba od liczby_naturalne() {
+                    jeśli(w > m) przerwij
+                    wynik = wynik + x*y*w*(z+1)
+                }
+            }
+        }
+    }
+    //(1+2+3+..+m)^4
```
Comment (ignacy123) [inline tests/generators/stacked_foreach.bzz:25]: I didn't check the formula behind this code, but I assume you got it right :smile:

**Review by ignacy123 — CHANGES_REQUESTED** [review]
Let's merge this after the other parts, so that the tests can pass

File: tests/generators/local_simple.bzz [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// INTEGRATION TESTS IGNORE
+przekaźnik p()-> Liczba {
+    przekaż 1
+}
+
+czynność główna() {
+    otrzymując x: Liczba od p(n)
```
[the remark this answers is not in the document]
Comment (ggawryal) [reply tests/generators/local_simple.bzz]: Yeah, it should be simply `p()`

File: tests/generators/stacked_foreach.bzz (hunk 2 above)
Comment (ggawryal) [reply tests/generators/stacked_foreach.bzz:25]: Actually, it's `1*nested_loop_result + 2*nested_loop_result + ... = (1+2+..+m)*nested_loop_result` :smiley:

File: tests/generators/mutual_recursion.bzz [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            zakończ
+        }
+        otrzymując x: Liczba od fib_przekaźnik(a-1) {
+            przekaż x+fib_czynność(a-2)
+            przerwij
+        }
+    }
+
+    zm wynik: Liczba = 0
+    otrzymując x: Liczba od fib_przekaźnik(a-1) {
```
[the remark this answers is not in the document]
Comment (ggawryal) [reply tests/generators/mutual_recursion.bzz:19]: You're right, it has been written incorrectly

**Comment by ggawryal** [comment]
I've run test on #165 and they are passing. However, I found an unrelated bug with binary negation inside a conditional statement - its intermediate form looks correct, but asm code is wrong - I've added ignored e2e test for that and it's currently failing.

**Review by ignacy123 — APPROVED** [review]

**Pull request merged**